### PR TITLE
I18n and farsi l10n

### PR DESCRIPTION
**Pull request by hkalbasi**
Here is the result: https://hkalbasi.github.io/breaklock/fa/

I tried to make changes as little as possible in the code, But I removed build results from master branch. So github pages from master branch (current state) wouldn't work. The `npm run build` now will create outputs in `public` folder and with `npm run deploy` it will push content of public folder in `gh-pages` branch and github pages will be updated (like my account).

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -45,24 +46,59 @@
...
-        <p>Good luck!_</p>
-        <p class="small">by <a href="https://twitter.com/mxwllt"><svg class="icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#icon-maxwellito"></use></svg></a> / on <a href="https://github.com/maxwellito/breaklock"><svg class="icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#icon-github"></use></svg></a></p>
+        <p>#@description_difficulty</p>
+        <p>#@label_goodluck</p>
+        <p class="small">#@attribution</p>
       </div>
     </div>
 
+    <!-- language selector -->
+    <div id="lang-selector" class="disabled">
```
Comment (maxwellito) [inline index.html]: Do you mind if we move the lang selector to a component like the other bits of the app?

File: package.json [hunk 2]
```diff
@@ -17,15 +18,21 @@
...
   "devDependencies": {
-    "babel-core": "^6.24.0",
-    "babel-loader": "^6.4.1",
-    "babel-preset-es2015": "^6.24.0",
+    "@babel/core": "^7.11.1",
+    "@babel/preset-env": "^7.11.0",
+    "babel-loader": "^8.1.0",
     "css-loader": "^0.28.0",
-    "extract-text-webpack-plugin": "^2.1.0",
+    "extract-text-webpack-plugin": "^3.0.2",
```
Comment (maxwellito) [inline package.json]: Is it still in use? If not, happy to ditch it :)

File: package.json [hunk 3]
```diff
@@ -17,15 +18,21 @@
...
     "node-sass": "^4.5.2",
     "raw-loader": "^0.5.1",
     "sass-loader": "^6.0.3",
     "style-loader": "^0.16.1",
-    "webpack": "^2.3.3"
+    "webpack": "^4.44.1"
+  },
+  "dependencies": {
+    "fs-extra": "^9.0.1",
+    "yaml": "^1.10.0"
```
Comment (maxwellito) [inline package.json:33]: `yaml` and `fs-extra` should be in dev dependencies, right?

**Review by maxwellito — COMMENTED** [review]
Hi @HKalbasi

First, thanks for this 'gigantic' PR. I must admit this is the biggest one I ever received on one of my repos.
I saw it last night but I couldn't review it. It's pretty impressive. 

You translated literally everything, even my terrible jokes. I mean, I guess because except asking Google translate, I won't be able to do much by myself.

- We should add the `lang` attribute on the `<html>` tag?
- Is there a way to use user preferences to redirect to the prefered language? (when landing on `breaklock/`) because I didn't see some piece of code responsible for it.
- I see the links the files path are part of the build process (`./` | `../`). Would it be more helpful to have a [`<base>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)?
- Is the service-worker + caching still working?

I will pull it on my machine an test it when I have some spare time.
But thanks for your PR, that was an awesome and incredible job!

- Max

**Comment by hkalbasi** [comment]
>     * We should add the `lang` attribute on the `<html>` tag?
Yes, And also `lang` in manifest. I will fix it now.
>     * Is there a way to use user preferences to redirect to the prefered language? (when landing on `breaklock/`) because I didn't see some piece of code responsible for it.
It can be redirected or notified that is available on her language, with js based on `navigator.languages`.

>     * I see the links the files path are part of the build process (`./` | `../`). Would it be more helpful to have a [`<base>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)?
It is much better. I did not know it.
>     * Is the service-worker + caching still working?
It should. At first time loading of a language, that language (and common assets) will be cached.

File: index.html (hunk 1 above)
Comment (hkalbasi) [reply index.html]: It makes sense. I botched it up :)

File: package.json (hunk 2 above)
Comment (hkalbasi) [reply package.json]:
```suggestion
```

**Comment by maxwellito** [comment]
Sounds awesome! Please tell me if you need any help on anything :)

**Comment by hkalbasi** [comment]
Done.

**Comment by maxwellito** [comment]
Thanks @HKalbasi ! The changes looks great! Im sorry about the delay, I'm a bit busy at the moment. I will build it on my machine to have little check in local, then it should be good to ship! Thanks again for your big PR!
I keep you posted :)

**Comment by hkalbasi** [comment]
No rush. Feel free :)

You're welcome! And thanks first for this nice game and second for the time you put on this PR.

**Review by maxwellito — APPROVED** [review]
👍

**Comment by maxwellito** [comment]
It's live!
Thanks again @HKalbasi for this wonderful work! 👍

**Pull request merged**